src/MagnumPlugins/GltfImporter/GltfImporter.cpp

```diff
@@ -3286,15 +3286,40 @@ Containers::Optional<MeshData> GltfImporter::doMesh(const UnsignedInt id, Unsign
         return {};
     }
 
+    /* Morph target attributes */
+    const Utility::JsonToken* gltfTargets = gltfPrimitive.find("targets"_s);
+    if(gltfTargets) {
+        if(!_d->gltf->parseArray(*gltfTargets)) {
+            Error{} << "Trade::GltfImporter::mesh(): invalid targets property in mesh" << _d->gltfMeshPrimitiveMap[id].first();
+            return {};
+        }
+
+        for(Utility::JsonArrayItem gltfTarget: gltfTargets->asArray()) {
+            if(!_d->gltf->parseObject(gltfTarget)) {
+                Error{} << "Trade::GltfImporter::mesh(): invalid morph target" << gltfTarget.index() << "in mesh" << _d->gltfMeshPrimitiveMap[id].first();
+                return {};
+            }
+
+            for(Utility::JsonObjectItem gltfMorphAttribute: gltfTarget.value().asObject()) {
+                if(!_d->gltf->parseUnsignedInt(gltfMorphAttribute.value())) {
+                    Error{} << "Trade::GltfImporter::mesh(): invalid morph target attribute" << gltfMorphAttribute.key() << "in mesh" << _d->gltfMeshPrimitiveMap[id].first();
+                    return {};
+                }
+
+                arrayAppend(attributeOrder, InPlaceInit, gltfMorphAttribute.key(), gltfMorphAttribute.value().asUnsignedInt(), gltfTarget.index());
+            }
+        }
```

Comment:
The `Trade::MeshAttributeData` constructor asserts that the morph target [isn't a weights, joints or object ID attribute](https://doc.magnum.graphics/magnum/namespaceMagnum_1_1Trade.html#a8a0ec4599a941c8f155bd76efc765808), so the importer has to check that and fail the import in that case.

Best place to do that would probably be in the if cascade below that translates the string names to `MeshAttribute`.

Reply:
I've updated it to behave similarly to the unsupported vertex format case. So it can still get imported as a custom attribute. Not entirely happy about the resulting error message, as it indicates the format being unsupported for said attribute, which isn't incorrect per se, but misleading in the sense that it could imply there exists a valid format (instead of the entire attribute not being supported in morph targets).

Reply:
I ended up making this an import error, as I don't expect these to be hit much, if at all, in practice, as they're disallowed by glTF anyway. Originally the unsupported vertex formats were also an import error, I just made those non-fatal after discovering there actually _are_ such assets in the wild, in particular non-normalized 16-bit vertex colors. So if something similar happens for morph targets, I can reconsider, but until then I'm keeping it simple and just failing.

The case where an unsupported format is in an otherwise supported morph target attribute (such as those non-normalized colors) is still handled as custom attribute, that stays.